files/cluster_provision.yml

```diff
@@ -66,34 +66,41 @@
                                 kernel_opts="ip=dhcp text inst.stage2=http://releases.linaro.org/reference-platform/enterprise/18.06/centos-installer/ inst.repo=http://mirror.centos.org/altarch/7/os/aarch64/ inst.ks=file:/ks.cfg"
                         fi
 
-                        # Chose known good kernel/initrd/cmdline
-
-                        # Build trigger machine_provision job
-                        cat << EOF > machine_provision
-                        machine_list=${machine_list}
-                        os_type=${os_type}
-                        machine_type=${machine_type}
+                        # Defaults
+                        machine_list="$( echo "$machine_list"| tr -d '[:space:]')"
```

Comment:
It doesn't make sense for you to hard-code the list with spaces up there, not use it and then trim them here. Why not create them without spaces to begin with?